module/zfs/dmu_send.c

```diff
@@ -2971,7 +2975,6 @@ receive_spill(struct receive_writer_arg *rwa, struct drr_spill *drrs,
 		dmu_tx_abort(tx);
 		return (err);
 	}
-	dmu_buf_will_dirty(db_spill, tx);
```

Comment:
Just to be clear, `dmu_buf_will_dirty()` needed to be removed because it unconditionally sets `DB_RF_MUST_SUCCEED` correct?

Reply:
We need to remove it because (1) it is unnecessary and (2) it calls `dbuf_read()` without the `NO_DECRYPT` flag, which wont work.